packages/vitest/src/node/plugins/conditions.ts

```diff
@@ -0,0 +1,32 @@
+function normalizeConditions(conditions?: string[]) {
+  if (!conditions) {
+    return []
+  }
+
+  // We remove browser and node conditions from the default list because
+  // we add back "node" explicitly in both client and server conditions.
+  // this ensures that the conditions we pass on to vite doesn't include
+  // both browser and node at the same time, or node twice in the server case.
+  return conditions.filter(c => c !== 'browser' && c !== 'node')
```

Comment:
I don't love that the vite migration docs say:
> For example, if you previously specified ['custom'] for resolve.conditions, you need to specify ['custom', ...defaultClientConditions] instead.

which implies that we should just be doing:
```ts
clientConditions: ['node', ...defaultClientConditions],
```
but there is some subtlety in that in vite 5, 'node' and 'browser' weren't actually part of the default conditions, but instead it passed:
```ts
browser: targetWeb && !additionalConditions.has('node')
```

So in the case of vitest passing node as an additional condition it would disable the browser resolve, but if node wasn't passed in, it would infer 'browser'.

If you want the same behavior as before, we need to remove "browser" from vite's default conditions, and add the node condition instead.